src/features/NoteEditor/RichEditor/RichEditorContent.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useRef } from 'react';
+import { Box, BoxProps, useMultiStyleConfig } from '@chakra-ui/react';
+import { CheckListPlugin } from '@lexical/react/LexicalCheckListPlugin';
+import { useLexicalComposerContext } from '@lexical/react/LexicalComposerContext';
+import { ContentEditable } from '@lexical/react/LexicalContentEditable';
+import { LexicalErrorBoundary } from '@lexical/react/LexicalErrorBoundary';
+import { HistoryPlugin } from '@lexical/react/LexicalHistoryPlugin';
+import { HorizontalRulePlugin } from '@lexical/react/LexicalHorizontalRulePlugin';
+import { LinkPlugin } from '@lexical/react/LexicalLinkPlugin';
+import { ListPlugin } from '@lexical/react/LexicalListPlugin';
+import { RichTextPlugin } from '@lexical/react/LexicalRichTextPlugin';
+import { TabIndentationPlugin } from '@lexical/react/LexicalTabIndentationPlugin';
+import { TablePlugin } from '@lexical/react/LexicalTablePlugin';
+
+import { CodeHighlightPlugin } from './plugins/CodeHighlightPlugin';
+import { GenericContextMenu } from './plugins/ContextMenu/components/GenericContextMenu';
+import { ContextMenuPlugin } from './plugins/ContextMenu/ContextMenuPlugin';
+import { EditorPanelPlugin } from './plugins/EditorPanelPlugin/EditorPanelPlugin';
+import { DropFilesPlugin } from './plugins/Files/DropFilesPlugin';
+import { FilesPlugin } from './plugins/Files/FilesPlugin';
+import { FormattingPlugin } from './plugins/Formatting/FormattingPlugin';
+import ImagesPlugin from './plugins/Image/ImagesPlugin';
+import { LinkClickHandlerPlugin } from './plugins/LinkClickHandlerPlugin';
+import {
+	MarkdownSerializePlugin,
+	MarkdownSerializePluginProps,
+} from './plugins/Markdown/MarkdownSerializePlugin';
+import { MarkdownShortcutPlugin } from './plugins/Markdown/MarkdownShortcutPlugin';
+
+export type RichEditorContentProps = BoxProps &
+	MarkdownSerializePluginProps & {
+		placeholder?: string;
+	};
+
+export const RichEditorContent = ({
+	value,
+	onChanged,
+	placeholder,
+	...props
+}: RichEditorContentProps) => {
+	const [editor] = useLexicalComposerContext();
+	const styles = useMultiStyleConfig('RichEditor');
+
+	const containerRef = useRef<HTMLDivElement | null>(null);
+	const isActive = () => {
+		const container = containerRef.current;
+		if (!container || !document.activeElement) return false;
+		if (
+			container !== document.activeElement &&
+			!container.contains(document.activeElement)
+		)
+			return false;
+
+		return true;
+	};
+
+	// toggle `editable` based on focus
+	useEffect(() => {
+		const onFocus = () => {
+			const editable = isActive();
+			console.log('Change editable state', editable);
+			editor.setEditable(editable);
```

Comment:
Find another way how to prevent auto focus on editor by change content. But do not change editable state